core/src/wallet/bitcoin/PersistentUTXOSource.cpp

```diff
@@ -0,0 +1,42 @@
+#include <bytes/serialization.hpp>
+#include <wallet/bitcoin/PersistentUTXOSource.hpp>
+#include <database/BlockchainDB.hpp>
+
+namespace ledger {
+    namespace core {
+        namespace bitcoin {
+            PersistentUTXOSource::PersistentUTXOSource(
+                const std::shared_ptr<db::BlockchainDB>& blockchainDB,
+                const std::shared_ptr<UTXOSourceInMemory>& inMemorySource
+            ) : _db(blockchainDB), _inMemorySource(inMemorySource) {
+            }
+
+            Future<UTXOSourceList> PersistentUTXOSource::getUTXOs(std::shared_ptr<api::ExecutionContext> ctx) {
+                auto self = shared_from_this();
+
+                return _inMemorySource->getUTXOs(ctx).template flatMap<UTXOSourceList>(ctx, [=](const UTXOSourceList& sourceList) {
+                    return _db->GetLastBlock().template map<UTXOSourceList>(ctx, [=](const Option<std::pair<uint32_t, db::ReadOnlyBlockchainDB::RawBlock>>& lastBlock) {
+                        if (lastBlock.hasValue()) {
+                            auto dbHeight = std::get<0>(*lastBlock);
+                            auto inMemHeight = self->_inMemorySource->currentSynchronizedHeight();
```

Comment:
currentSynchronizedHeight should be returned in the getUTXOs, somebody can call getUTXOs in other thread just after our getUTXOs call but before currentSynchronizedHeight(), that may change the value of currentSynchronizedHeight .
In general I think we will have the height returned with actual result in a lot of functions, I don't see other thread-safe mechanism.

Reply:
Done.